sssd_test_framework/utils/sss_override.py

```diff
@@ -0,0 +1,404 @@
+        """
+
+        args: CLIBuilderArgs = {
+            "name": (self.util.cli.option.VALUE, name),
+            "uid": (self.util.cli.option.VALUE, uid),
+            "gid": (self.util.cli.option.VALUE, gid),
+            "home": (self.util.cli.option.VALUE, home),
+            "gecos": (self.util.cli.option.VALUE, gecos),
+            "shell": (self.util.cli.option.VALUE, shell),
+            "certificate": (self.util.cli.option.VALUE, certificate),
+        }
+
+        self.util.logger.info(f"Modifying local override for user {self.user} on {self.util.host.hostname}")
+        self.util.host.ssh.exec(["sss_override", "user-add", self.user] + self.util.cli.args(args))
+
+        return self
+
+    def delete(self) -> SSSOverrideUser:
+        """
+        Delete the local override for user.
+        """
+        self.util.logger.info(f"Deleting local override for user {self.user} on {self.util.host.hostname}")
+        self.util.host.ssh.exec(["sss_override", "user-del", self.user], raise_on_error=False)
+
+        return self
+
+    def backup(
+        self,
+        file: str = "/tmp/sss_override_user.bak",
+    ) -> SSSOverrideUser:
+        """
+        Exports local override data for **all** users.
+
+        :param file: File name.
+        :type file: str | None, defaults to "/tmp/sss_override_user.bak"
+        :return: Self.
+        :rtype: SSSOverrideUser
+        """
+        self.util.logger.info(f"Exporting local override user data to {file} on {self.util.host.hostname}")
+        self.util.fs.backup(file)
+        self.util.host.ssh.exec(["sss_override", "user-export", file])
+
+        return self
+
+    def restore(
+        self,
+        file: str = "/tmp/sss_override_user.bak",
+    ) -> SSSOverrideUser:
+        """
+        Imports local override data from file
+
+        :param file: File name.
+        :type file: str | None, defaults to "/tmp/sss_override_user.bak"
+        :return: Self.
+        :rtype: SSSOverrideUser
+        """
+        self.util.logger.info(f"Importing local override user data from {file} on {self.util.host.hostname}")
+        self.util.host.ssh.exec(["sss_override", "user-import", file])
+
+        return self
+
+    def get(self, attrs: list[str] | None = None) -> dict:
+        """
+        Get local override data for user.
+
+        :param attrs: If set, only requested attributes are returned, defaults to None, returning all attributes
+        :type attrs: list[str] | None, optional
+        :return: Dictionary with attribute name as a key.
+        :rtype: dict[str, list[str]]
+        """
+
+        self.util.logger.info(f"Fetching local override data for user {self.user} on {self.util.host.hostname}")
+        output = self.util.host.ssh.exec(["sss_override", "user-show", self.user])
+        lst = output.stdout.split(":")
+        try:
+            assert len(lst) == 7
+        except IndexError:
+            print("Unexpected index value, unable to get user data")
```

Comment:
Avoid using assert outside test. This should be:

```python
if len(lst) != 7:
    raise ValueError(f"Unexpected output: {output.stdout}")
```

`assert len(lst) == 7` will not even raise `IndexError` but `AssertionError` so that is a dead code. And if you catch the exception youjust print it to stdout but the test would continue. We want to raise exception so the test fails.

Reply:
Okay, I understand what I didn't do, I didn't actually raise an exception.